### PR TITLE
Replace buoy_compiler with foil

### PR DESCRIPTION
**Pull request by rkallos**
This replaces `buoy_compiler` with foil, which does basically the same thing as `buoy_compiler` did (fast KV lookup using compiled Erlang modules), but using the same lib as shackle & co.

### CODE REVIEW

File: rebar.config.script [hunk 1]
```diff
@@ -3,6 +3,8 @@ Config = case erlang:function_exported(rebar3, main, 1) of
         CONFIG;
     false ->
         [{deps, [
+            {foil, ".*",
+                {git, "https://github.com/lpgauth/foil.git", {tag, "0.2.1"}}},
```
Comment (lpgauth) [inline rebar.config.script]: `/0.2.1/0.1.2/`

Comment (rkallos) [reply rebar.config.script]: ...whoops...

**Comment by rkallos** [comment]
I don't know how to fix the Travis complaint. :cry: 

EDIT: It looks like there was an incompatibility with `erl_tar` since OTP 20. The Foil package on hex.pm was created using OTP 20, and for some reason couldn't be backported for use with OTP 18 and 19. Fetching and compiling foil from Github fixed this.

**Comment by lpgauth** [comment]
Looks good, thanks!

**Pull request merged**